src/main/java/tc/oc/pgm/rotation/MapPoolManager.java

```diff
@@ -0,0 +1,157 @@
+package tc.oc.pgm.rotation;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.*;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import org.apache.commons.io.FileUtils;
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
+import org.bukkit.configuration.file.FileConfiguration;
+import org.bukkit.configuration.file.YamlConfiguration;
+import tc.oc.pgm.AllTranslations;
+import tc.oc.pgm.api.PGM;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.map.PGMMap;
+
+/**
+ * Manages all the existing {@link MapPool}s, as for maintaining their order, and updating the one
+ * {@link PGM} will use after every match depending on the player count (Dynamic Rotations)
+ */
+public class MapPoolManager implements PGMMapOrder {
+
+  private Logger logger;
+
+  private File mapPoolsFile;
+  private FileConfiguration mapPoolFileConfig;
+  private List<MapPool> mapPools = new ArrayList<>();
+  private MapPool activeMapPool;
+  /** When a {@link PGMMap} is manually set next, it overrides the rotation order * */
+  private PGMMap overriderMap;
+
+  public MapPoolManager(Logger logger, File mapPoolsFile) {
+    this.mapPoolsFile = mapPoolsFile;
+    this.logger = logger;
+
+    if (!mapPoolsFile.exists()) {
+      try {
+        FileUtils.copyInputStreamToFile(PGM.get().getResource("map-pools.yml"), mapPoolsFile);
+      } catch (IOException e) {
+        logger.log(Level.SEVERE, "Failed to create the map-pools.yml file", e);
+        return;
+      }
+    }
+
+    this.mapPoolFileConfig = YamlConfiguration.loadConfiguration(mapPoolsFile);
+    loadMapPools();
+  }
+
+  private void loadMapPools() {
+    mapPools =
+        mapPoolFileConfig.getConfigurationSection("pools").getKeys(false).stream()
+            .map(key -> MapPool.of(this, mapPoolFileConfig, key))
+            .filter(MapPool::isEnabled)
+            .collect(Collectors.toList());
+
+    MapPool lastActiveMapPool = getMapPoolByName(mapPoolFileConfig.getString("last_active"));
+    setActiveMapPool(lastActiveMapPool);
+  }
+
+  public void saveMapPools() {
+    try {
+      mapPoolFileConfig.save(mapPoolsFile);
+    } catch (IOException e) {
+      logger.log(Level.SEVERE, "Could not save next map for future reference", e);
+    }
+  }
+
+  private void setActiveMapPool(MapPool activeMapPool) {
+    this.activeMapPool = activeMapPool;
+  }
+
+  public MapPool getActiveMapPool() {
+    return activeMapPool;
+  }
+
+  public List<MapPool> getMapPools() {
+    return mapPools;
+  }
+
+  private void updateActiveMapPool(MapPool mapPool) {
+    if (mapPool == activeMapPool) return;
+
+    setActiveMapPool(mapPool);
+
+    mapPoolFileConfig.set("last_active", activeMapPool.getName());
+    saveMapPools();
+
+    Bukkit.broadcastMessage(
```

Comment:
How about `MatchManager#getMatches()#sendMessage` ?

Reply:
I inherited this from older code, i'll actually pass the proper match to here and broadcast to the players on it